static/js/components/EducationTab.js

```diff
@@ -0,0 +1,249 @@
+import _ from 'lodash';
+import {
+  boundMonthField,
+  boundYearField,
+  saveAndContinue
+} from '../util/profile_edit';
+import {
+  toggleEducationLevel,
+  openEducationForm,
+  closeEducationForm
+} from '../actions';
+import { HIGH_SCHOOL, ASSOCIATE, BACHELORS, MASTERS, DOCTORATE } from '../constants';
+import { Grid, Cell, Switch, Dialog, DialogTitle, DialogContent, DialogActions, FABButton, Icon } from 'react-mdl';
+
+class EducationTab extends ProfileTab {
+  constructor(props) {
+    super(props);
+    this.boundMonthField = boundMonthField.bind(this);
+    this.boundYearField = boundYearField.bind(this);
+    this.educationLevelOptions = [
+      {value: HIGH_SCHOOL, label: "High school"},
+      {value: ASSOCIATE, label: 'Associate degree'},
+      {value: BACHELORS, label: "Bachelor's degree"},
+      {value: MASTERS, label: "Master's or professional degree"},
+      {value: DOCTORATE, label: "Doctorate"}
+    ];
+
+    this.educationLevelLabels = {};
+    this.educationLevelOptions.forEach(level => {
+      this.educationLevelLabels[level.value] = level.label;
+    });
+    this.saveAndContinue = saveAndContinue.bind(this, '/profile/professional', EducationTab.validation);
+    this.openNewEducationForm = this.openNewEducationForm.bind(this);
+    this.handleCloseDialog = this.handleCloseDialog.bind(this);
+
+  }
+  static propTypes ={
+    educationLevels: React.PropTypes.object,
+    educationDialog: React.PropTypes.object,
+    profile: React.PropTypes.object,
+    saveProfile: React.PropTypes.func,
+    updateProfile: React.PropTypes.func,
+    errors: React.PropTypes.object,
+    dispatch: React.PropTypes.func.isRequired
+  };
+
+  static defaultProps = {
+    requiredFields: [],
+    validationMessages: {
+      'degree_name': "'Degree level",
+      'graduation_date': 'Graduation Date',
+      'field_of_study': 'Field of study',
+      'online_degree': 'Online Degree',
+      'school_name': 'School name',
+      'school_city': 'City',
+      'school_state_or_territory': 'State',
+      'school_country': 'Country'
+    }
+  };
+
+  static validation(profile, requiredFields) {
+    const keys = [
+      'degree_name',
+      'graduation_date',
+      'field_of_study',
+      'online_degree',
+      'school_name',
+      'school_city',
+      'school_state_or_territory',
+      'school_country'
+    ];
+    let nestedFields = (index) => {
+      let keySet = (key) => ['education', index, key];
+      return keys.map(key => keySet(key));
+    };
+    return requiredFields.concat(
+      ...profile.education.map((v, i) => nestedFields(i))
+    );
+  }
+
+  openNewEducationForm(level, index) {
+    const {dispatch, profile, updateProfile} = this.props;
+    let new_index = index;
+    if (index === null){
+      new_index = profile['education'].length;
+    }
+    /* add empty education */
+    let clone = Object.assign({}, profile);
+    clone['education'].concat(generateNewEducation(level));
```

Comment:
I think you are right. Not sure why George asked me to replace push with concat.